game/navigator/navigator_service.go

```diff
@@ -18,7 +18,7 @@ type navService struct {
 // NavigatorService will initialize the single instance of navService if it is the first time it is called and then it
 // will return the instance.
 func NavigatorService() *navService {
-	nonce.Do(func() {
+	once.Do(func() {
```

Comment:
Avoid calling this function as global, instead inject this service where needed and pass it around as local variable in structs.